mode_farm_generic.lua

```diff
@@ -39,44 +36,52 @@ function GetDesire()
 		return heroDesire
 	end
 
-	if #EnemyHeroes > 0 then
-		return BOT_MODE_DESIRE_NONE;
-	end
-
-	if not bot:IsAlive() or bot:IsChanneling() or bot:GetCurrentActionType() == 1 or bot:GetNextItemPurchaseValue() == 0
-		or bot:WasRecentlyDamagedByAnyHero(3.0) or #EnemyHeroes >= 1
+	-- Requirements to allow farming mode:
+	-- * Be alive
+	-- * Be a hero that farms
+	-- * Between level 6 and 30
+	-- * No humans on team
+	-- * Game is not mid only
+	-- * Have something to buy
+	-- * No enemies nearby, and none have damaged us recently
+	-- * No enemy heroes near our base
+	-- * We aren't trying to retreat or go to secret shop
+
+	if not (bot:IsAlive()
+			and campUtils.IsStrongJungler(bot)
+			and bot:GetLevel() >=6 and bot:GetLevel() < 30)
+		or IsHumanPlayerInTeam()
```

Comment:
Is this condition necessary? I don't quite understand the difference when human players are present.

Reply:
That logic was already there and I assume working as intended.  I'm guessing the intent is that the human player may want to farm, and the bot farming as well could interfere with that.  That check can be removed if unwanted; nothing else depends on it.

Reply:
@JoshdanG Now we only have 2 heroes with farming mode enabled? If so, it is indeed possible to remove this condition, it affects human players less.